tests/testthat/test-update_values.R

```diff
@@ -2,44 +2,81 @@
 
 x2 = data.table(id = c(1, 1, 2, 3, NA),
                  t = c(1L, 2L, 1L, 2L, NA_integer_),
-                 x = c(16, 12, NA, NA, 15))
+                 x = c(16, 12, NA, NA, 15),
+                y = c(NA, 2, 3, 4, 5))
 
 y2 = data.table(id = c(1, 2, 5, 6, 3),
                 yd = c(1, 2, 5, 6, 3),
                 y  = c(11L, 15L, 20L, 13L, 10L),
-                x  = c(16:20))
-
-test_that("update_values works as expected", {
-
+                x  = c(NA, 17:20))
   # Joined data frame with values in x2 vars not updated
   dt = joyn(x2,
             y2,
             by         = "id",
             match_type = "m:1",
             update_NAs = FALSE,
-            keep_common_vars = TRUE)
+            update_values = FALSE,
+            keep_common_vars = TRUE,
+            reporttype = "numeric",
+            verbose = FALSE)
+
+test_that("update_values of one variable", {
 
-  colnames(dt)[3] = "x"
 
   # Updating values in x2$x with values from y2$x
   res <- update_values(dt,
                        var       = "x",
                        reportvar = ".joyn")
+  # check not updated values
+  which(is.na(res$x.x) & res$.joyn != 2) |>
```

Comment:
No values in x should be updated here for dt. Because every row in x either has an NA or the corresponding row in y has an NA.